.github/workflows/amd-build-ci.yml

```diff
@@ -0,0 +1,57 @@
+name: amd-build-ci
+run-name: ${{ github.actor }} is kicking off a ROCm build
+on: pull_request
+jobs:
+  rocm-build:
+    runs-on: [self-hosted, amd]
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          export ROCM_PATH=/opt/rocm-5.5.0
+          export PATH=${ROCM_PATH}/bin:${ROCM_PATH}/llvm/bin:${PATH}
+          SRCROOT=`pwd`
+          BUILDROOT=`mktemp -d build-XXXXXXXX`
+          INSTALLROOT=`mktemp -d install-XXXXXXXX`
+          QUDA_GPU_ARCH=gfx90a
+          cmake ${SRCROOT} \
+            -B ${BUILDROOT} \
+            -DQUDA_TARGET_TYPE="HIP" \
+            -DQUDA_GPU_ARCH=${QUDA_GPU_ARCH} \
+            -DROCM_PATH=${ROCM_PATH} \
+            -DQUDA_DIRAC_CLOVER=ON \
+            -DQUDA_DIRAC_CLOVER_HASENBUSCH=OFF \
+            -DQUDA_DIRAC_DOMAIN_WALL=OFF \
+            -DQUDA_DIRAC_NDEG_TWISTED_MASS=OFF \
+            -DQUDA_DIRAC_STAGGERED=ON \
+            -DQUDA_DIRAC_TWISTED_MASS=OFF \
+            -DQUDA_DIRAC_TWISTED_CLOVER=OFF \
+            -DQUDA_DIRAC_WILSON=ON \
+            -DQUDA_CLOVER_DYNAMIC=ON \
+            -DQUDA_FORCE_HISQ=ON \
+            -DQUDA_QDPJIT=OFF \
+            -DQUDA_INTERFACE_QDPJIT=OFF \
+            -DQUDA_INTERFACE_MILC=ON \
+            -DQUDA_INTERFACE_CPS=OFF \
+            -DQUDA_INTERFACE_QDP=ON \
+            -DQUDA_INTERFACE_TIFR=OFF \
+            -DQUDA_QMP=ON \
+            -DQUDA_DOWNLOAD_USQCD=ON \
+            -DQUDA_OPENMP=OFF \
+            -DQUDA_MULTIGRID=ON \
+            -DQUDA_MAX_MULTI_BLAS_N=9 \
+            -DQUDA_DOWNLOAD_EIGEN=ON \
+            -DQUDA_PRECISION=8 \
```

Comment:
Depending on how much coverage you want, this would ideally be `14` (double, single and half precision), but if you only want one precision to be enabled `4` is best (this is pure single, for which nearly all kernels are compiled for)

Reply:
Removed.